src/hapi_schema/db_national_risk.py

```diff
@@ -20,31 +19,45 @@
 from hapi_schema.db_location import DBLocation
 from hapi_schema.db_resource import DBResource
 from hapi_schema.utils.base import Base
+from hapi_schema.utils.constraints import (
+    general_risk_constraint,
+    reference_period_constraint,
+)
+from hapi_schema.utils.enums import RiskClass
 from hapi_schema.utils.view_params import ViewParams
 
 
 class DBNationalRisk(Base):
     __tablename__ = "national_risk"
     __table_args__ = (
+        general_risk_constraint("overall"),
+        general_risk_constraint("hazard_exposure"),
+        general_risk_constraint("vulnerability"),
+        general_risk_constraint("coping_capacity"),
+        CheckConstraint(
+            "(global_rank >= 0) AND (global_rank <= 250)", name="global_rank"
+        ),
         CheckConstraint(
             "meta_avg_recentness_years >= 0.0",
             name="meta_avg_recentness_years",
         ),
         CheckConstraint(
-            "(reference_period_end >= reference_period_start) OR (reference_period_start IS NULL)",
-            name="reference_period",
+            "(meta_missing_indicators_pct >= 0.0) AND (meta_missing_indicators_pct <= 10.0)",
+            name="meta_missing_indicators_pct",
         ),
+        reference_period_constraint(),
     )
 
-    id: Mapped[int] = mapped_column(Integer, primary_key=True)
-    resource_hdx_id: Mapped[int] = mapped_column(
+    resource_hdx_id: Mapped[str] = mapped_column(
         ForeignKey("resource.hdx_id", onupdate="CASCADE", ondelete="CASCADE"),
         nullable=False,
     )
     admin2_ref: Mapped[int] = mapped_column(
-        ForeignKey("admin2.id", onupdate="CASCADE"), nullable=False
+        ForeignKey("admin2.id", onupdate="CASCADE"), primary_key=True
+    )
```

Comment:
I think while we're changing everything we should change admin2_ref to location_ref.  This table will never have subnational data so we should just link to the countries, rather than the unspecified admin1 and admin2 units.

Reply:
Oops actually it's supposed to be location_ref according to the schema, thanks for catching this!